File: src/main/java/org/example/kalkulationsprogramm/domain/LieferantBild.java
package org.example.kalkulationsprogramm.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

/**
 * Bilddatei, die zu einem Lieferanten gehört.
 * Verwendet für Reklamationsbilder und andere Dokumentation.
 */
@Getter
@Setter
@Entity
public class LieferantBild {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "lieferant_id", nullable = false)
    private Lieferanten lieferant;

    @Column(nullable = false)
    private String originalDateiname;

    @Column(nullable = false)
    private String gespeicherterDateiname;

    private String beschreibung;

    @Column(name = "erstellt_am")
    private LocalDateTime erstelltAm;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "mitarbeiter_id")
    private Mitarbeiter hochgeladenVon;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reklamation_id")
    private LieferantReklamation reklamation;
}

File: src/main/kotlin/org/example/kalkulationsprogramm/domain/LieferantBild.kt
package org.example.kalkulationsprogramm.domain

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
open class LieferantBild {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "lieferant_id", nullable = false)
    open var lieferant: Lieferanten? = null

    @Column(nullable = false)
    open var originalDateiname: String? = null

    @Column(nullable = false)
    open var gespeicherterDateiname: String? = null

    open var beschreibung: String? = null

    @Column(name = "erstellt_am")
    open var erstelltAm: LocalDateTime? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "mitarbeiter_id")
    open var hochgeladenVon: Mitarbeiter? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reklamation_id")
    open var reklamation: LieferantReklamation? = null

}